src/main/scala/com/tapad/docker/DockerComposePlugin.scala

```diff
@@ -362,12 +363,14 @@ class DockerComposePluginLocal extends AutoPlugin with ComposeFile with DockerCo
       print(s"Inspecting container $containerId to get the port mappings")
       val containerInspectInfo = getDockerContainerInfo(containerId)
       val jsonInspect = parse(containerInspectInfo)
-      //For each internal container port find its externally mapped host accessible port
-      val portsWithHost = service.ports.map { port =>
-        //If not specified assume it's a tcp port
-        val portFullName = if (port.containerPort.contains("/")) port.containerPort else s"${port.containerPort}/tcp"
-        val hostPort = compact(render(jsonInspect \ "NetworkSettings" \ "Ports" \ portFullName \ "HostPort")).replaceAll("\"", "")
-        PortInfo(hostPort, port.containerPort, port.isDebug)
+
+      val exposedPorts = getDockerPortMappings(containerId)
+      val portMappingPattern = """(\d+).*:(\d+)""".r
+      val portsWithHost = {
+        exposedPorts.split("\n").toList.map {
+          case portMappingPattern(container, host) =>
+            PortInfo(host, container, Try(service.ports.find(_.containerPort == container).get.isDebug).getOrElse(false))
```

Comment:
You can probably clean up the debug port detection a bit: `service.ports.exists(p => p.containerPort == container && p.isDebug)`